--- cvxopt-1.3.3/setup.py.orig	2026-03-12 22:36:38
+++ cvxopt-1.3.3/setup.py	2026-03-12 22:36:42
@@ -186,6 +186,9 @@
     extra_link_args = BLAS_EXTRA_LINK_ARGS,
     sources = ['src/C/lapack.c'] )
 
+if sys.maxsize > 2**31:
+    MACROS += [('DLONG', None)]
+
 if not SUITESPARSE_SRC_DIR:
     umfpack = Extension('umfpack',
         libraries = ['umfpack','cholmod','amd','colamd','suitesparseconfig'] + LAPACK_LIB + BLAS_LIB + RT_LIB,
